lib/aws/rails/middleware/elastic_beanstalk_sqsd.rb

```diff
@@ -31,37 +31,42 @@ def call(env)
         private
 
         def execute_job(request)
-          # Jobs queued from the Active Job SQS adapter contain the JSON message in the request body.
-          job = Aws::Json.load(request.body.string)
+          # Jobs queued from the SQS adapter contain the JSON message in the request body.
+          job = ::ActiveSupport::JSON.decode(request.body.string)
           job_name = job['job_class']
           @logger.debug("Executing job: #{job_name}")
-
-          begin
-            ::ActiveJob::Base.execute(job)
-          rescue NameError => e
-            @logger.error("Job #{job_name} could not resolve to a class that inherits from Active Job.")
-            @logger.error("Error: #{e}")
-            return internal_error_response
-          end
-
+          _execute_job(job, job_name)
           [200, { 'Content-Type' => 'text/plain' }, ["Successfully ran job #{job_name}."]]
+        rescue NameError
+          internal_error_response
+        end
+
+        def _execute_job(job, job_name)
+          ::ActiveJob::Base.execute(job)
+        rescue NameError => e
+          @logger.error("Job #{job_name} could not resolve to a class that inherits from Active Job.")
+          @logger.error("Error: #{e}")
+          raise e
         end
 
         def execute_periodic_task(request)
           # The beanstalk worker SQS Daemon will add the 'X-Aws-Sqsd-Taskname' for periodic tasks set in cron.yaml.
           job_name = request.headers['X-Aws-Sqsd-Taskname']
           @logger.debug("Creating and executing periodic task: #{job_name}")
-
-          begin
-            job = job_name.constantize.new
-            job.perform_now
-          rescue NameError => e
-            @logger.error("Periodic task #{job_name} could not resolve to an Active Job class - check the spelling in cron.yaml.")
-            @logger.error("Error: #{e}.")
-            return internal_error_response
-          end
-
+          _execute_periodic_task(job_name)
```

Comment:
Where is the path being checked?

Reply:
Oh, I made an oopsie, but tests passed.

Reply:
It's checked in `periodic_task?` which only invokes the job by name. Otherwise, cron will just call a route, and the route can do whatever (out of scope for this middleware)